### ISSUE TITLE
can you write me the code for getting the score?

### ISSUE THREAD

**Issue by chrisfmontes** [arendsee/bcb660-project#1]
Hi! I hope you arrived without any issue.
I'm writing to ask if you could write that piece of code you showed me that grabbed the maximum scored read, and printed back the score and the accession number. To make the histogram.

**Comment by arendsee** [comment]
Ah, yes, everything is good here, I just got settled and connected. I'll post the histogram code in a minute.

**Comment by arendsee** [comment]
Aww shit, it looks like I didn't commit the stuff I did just before leaving. That `join-nd6-mito.sh` script is totally broken.

**Comment by chrisfmontes** [comment]
I just used the commands from the "one-gene-from-all" script, I wrote a R script for the last part but it takes forever

**Comment by arendsee** [comment]
Yeah, I imagine. R gets pretty slow as files approach the gigabyte size. I've just pushed a script for adding scientific names to blast results. I'll get the other script together soon.

Have you seen the `autoslurm.sh` script? I use it to automate submissions to slurm. Then you can just write a standard shellscript and pass it as an argument to `autoslurm`, where the shellscript will be wrapped and submitted. I don't like polluting my scripts with cluster specific boilerplate.

**Comment by arendsee** [comment]
OK, I've pushed a script for making the histogram. It also saves the species-max_bitscore table

**Comment by chrisfmontes** [comment]
Thanks, I'll try it over the weekend so we can get this done.
I didn't try the autoslurm script, because I couldn't understand the code.

**Comment by chrisfmontes** [comment]
Ok, so I ran the scripts and got the histogram, but now I want to do a different approach. I want to use the make-histogram script with the n=3 (column 3) argument (in order to get the maximum score per read instead of per specie. With this we can obtain the abundance distribution of species and select the species that are more represented in the BLAST output). But I want the output to still give me the score and the name of the specie (so I can make the abundance plot based on specie name). Can it be done? I know how to make it on R, but it takes forever.

**Comment by arendsee** [comment]
Apologies for the late reply.

Try this:

``` R
require(readr)
require(dplyr)
require(magrittr)

blastfile <- 'nd6-blast-with-names.tab'

d <- read_tsv(blastfile, col_names=FALSE, comment="#") %>%
  select(X2,X3,X8) %>%
  rename(species=X2,read=X3,bitscore=X8)
d$species <- gsub("_", " ", d$species)

# total number of species
(total <- length(unique(d$species)))

# species list (you can plug this into NCBI common tree)
write(unique(d$species), file="species-with-hits.txt")


# Get just the matches that are maximum for each read
d <- group_by(d, read) %>% dplyr::filter(bitscore == max(bitscore))

# number of species with maximum against at least one read
(withmax <- length(unique(d$species)))

(counts <- summary(factor(d$species), maxsum=Inf))
write(unique(d$species), file="species-with-a-max-hit.txt")

hist(log(counts))

names(counts[counts > 10]) %>%
  write(file="species-with-more-than-10-max-hits.txt")
```
There is a lot of fun stuff to see in here.

**Comment by chrisfmontes** [comment]
Thanks a lot! I'll try the code and let you know of the output.

**Comment by chrisfmontes** [comment]
Hey Zeb, Andrew just posted this on blackboard:
 Final Project

Posted on: Tuesday, May 2, 2017 3:49:20 PM CDT

This is just a reminder that final project reports are due on Friday this week.  I will not be accepting late submissions as I need time to grade them.  So be sure that your report is submitted by Friday.

Also, EVERY PERSON in the group needs to submit the final report with their interpretation of the results and MAJOR CONTRIBUTIONS to the report clearly indicated. 

If you were not able to get results, then please discuss the problems you encountered and the strategies you are working on to try to overcome these problems. 

If you want, I can write an email to let him know that you can't upload your version of the report on blackboard (I will still post the report here on github, once we have it ready)

**Comment by arendsee** [comment]
I actually can access blackboard. I can't access the cluster, though. I got slack working, we can chat there is you like.
